[feat] 안읽은 채팅 메세지 개수 확인 기능

pie2457 opened the pull request:
## Issues
- #66 

## What is this PR? 👓
읽지 않은 채팅방의 메세지 개수를 확인하는 기능에 대한 PR입니다.

## Key changes 🔑
- 이전 PR에 깜빡한 테스트 코드 수정
- 안읽은 채팅 메세지 개수 확인 기능
- 안읽은 채팅 메세지 개수 확인 기능 테스트

pie2457 committed:
refactor #64 : 리뷰사항 반영

- ChatLogServiceTest : 브랜치 64 PR에 깜빡하고 빼먹은 테스트 수정 -> 채팅의 개수를 늘려 0이후의 메세지를 모두 가져오는지 확인

pie2457 committed:
feat #66 : 안읽은 채팅 메세지 개수 확인 기능 추가

pie2457 committed:
test #66 : 안읽은 채팅 메세지 개수 확인 기능 테스트

pie2457 committed:
feat #66 : 메세지 읽음 처리 기능

getMessages시 newCount를 0으로 업데이트 시키는 기능

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/chat/ChatroomService.java:
```diff
@@ -37,6 +37,11 @@ public CustomSlice<ChatroomListResponse> findAll(Principal principal, int size,
 
 		List<ChatroomListResponse> content = responses.getContent();
 
+		for (ChatroomListResponse chatroomListResponse : content) {
+			Long messageCount = chatroomRepository.getNewMessageCount(chatroomListResponse.getChatroomId());
+			chatroomListResponse.assignNewMessageCount(messageCount);
+		}
```
해당 로직을 메서드로 분리하는 건 어떻게 생각하세요?
다른 사람이 볼 때는 해당 로직이 어떤 일을 하는지 직관적으로 알기 힘들 수도 있을 것 같아서요!

pie2457 replied:
넵 안그래도 추가 해야할 것 들이 있어서 다음 PR에 수정한 내용을 올릴 예정입니다 감사합니당 ~!

23Yong commented on src/main/java/pie/tomato/tomatomarket/domain/ChatLog.java:
```diff
@@ -47,6 +47,6 @@ private ChatLog(String message, String sender, String receiver, LocalDateTime cr
 	public static ChatLog of(PostMessageRequest request, Principal principal, String receiver, Chatroom chatroom) {
 		return new ChatLog(
-			request.getMessage(), principal.getNickname(), receiver, LocalDateTime.now(), 0L, chatroom);
+			request.getMessage(), principal.getNickname(), receiver, LocalDateTime.now(), 1L, chatroom);
```
1L이 어떤 의미인지 모를 수도 있을 것 같아요~ 상수처리 하는건 어떻게 생각하세요?

pie2457 replied:
넵 그러네요! 수정할께요 감사합니다 ~!

23Yong commented on src/main/java/pie/tomato/tomatomarket/infrastructure/persistence/chat/ChatPaginationRepository.java:
```diff
@@ -24,21 +24,20 @@ public class ChatPaginationRepository {
+	public Slice<ChatroomListResponse> findAll(Long memberId, int size, Long chatroomId) {
 		List<ChatroomListResponse> responses = queryFactory.select(Projections.fields(ChatroomListResponse.class,
-				chatLog.id.as("chatroomId"),
+				chatLog.chatroom.id.as("chatroomId"),
```
이런 버그가 있었군요.. 저도 더 꼼꼼하게 보겠습니당

pie2457 replied:
ㅋㅋ 아닙니당 감사해요 !!

23Yong commented on src/test/java/pie/tomato/tomatomarket/application/ChatroomServiceTest.java:
```diff
@@ -86,6 +86,31 @@ void registerChatroom() {
 
+	@DisplayName("안읽은 채팅 메세지 개수를 확인한다.")
+	@Test
+	void newMessageCount() {
```
`countUnreadMessageCount()` 같이 동사 네이밍은 어떠세요?

pie2457 replied:
넵 훨씬 좋네요 수정할게요 ! 감사합니당 ~~

23Yong reviewed (approved):
수고하셨어요~
채팅방관련 로직이 복잡해서 헷갈리는 부분이 많이 생기는 것 같네요!

pie2457 committed:
refactor #66 : 리뷰사항 반영

- ChatLog : 안읽은 메세지 카운트 수를 1L로 설정 -> 의미가 모호해서 상수로 처리
- ChatroomServiceTest : 메서드명 변경
newMessageCount() -> countUnreadMessage()
